builder/hspkg-builder.nix

```diff
@@ -57,8 +57,10 @@ let
             ;
   };
 
-in {
+in rec {
   components = haskellLib.applyComponents buildComp config;
+  run = builtins.mapAttrs (_: es: builtins.mapAttrs (_: d: d.run) es)
+    { inherit (components) exes tests benchmarks; };
```

Comment:
I can't really see why users would want to run exes and benchmarks like this.

It could be simpler with a single `test-runners` attribute.

Reply:
An exe might be pure generator (for instance it could build an html file or a log table).  I think making sure it runs could be meaningful test.

Checking that a benchmarks runs seems like it could also be a meaningful test.  You might not care about the performance result when run in nix (since it is hard to know what load the machine will be under), but you could also have it `exitFailure` if the result was very poor.